apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -1,311 +1,479 @@
         }
-        let provider = provider_builder.build();
-
-        let tracer = provider.versioned_tracer(
-            "apollo-opentelemetry",
-            Some(env!("CARGO_PKG_VERSION")),
-            None,
-        );
-        let _prev_global_provider = global::set_tracer_provider(provider);
-
-        Ok(tracer)
     }
 
-    /// Create a client to talk to our spaceport and return an exporter.
-    pub(crate) fn build_exporter(&self) -> Result<Exporter, ApolloError> {
-        let collector = match self.spaceport_config.clone() {
-            Some(cfg) => cfg.collector,
-            None => DEFAULT_SERVER_URL.to_string(),
+    fn extract_http_data(&self, span: &SpanData) -> Http {
+        let method = match span
+            .attributes
+            .get(&Key::new("method"))
+            .map(|data| data.as_str())
+            .unwrap_or_default()
+            .as_ref()
+        {
+            "OPTIONS" => 1,
+            "GET" => 2,
+            "HEAD" => 3,
+            "POST" => 4,
+            "PUT" => 5,
+            "DELETE" => 6,
+            "TRACE" => 7,
+            "CONNECT" => 8,
+            "PATCH" => 9,
+            _ => 0,
         };
-        let graph = self.graph_config.clone();
-
-        tracing::debug!("collector: {}", collector);
-        tracing::debug!("graph: {:?}", graph);
+        let version = span
+            .attributes
+            .get(&Key::new("version"))
+            .map(|data| data.as_str())
+            .unwrap_or_default();
+        let headers = span
+            .attributes
+            .get(&Key::new("headers"))
+            .map(|data| data.as_str())
+            .unwrap_or_default();
+        let uri = Uri::from_str(
+            &span
+                .attributes
+                .get(&Key::new("uri"))
+                .map(|data| data.as_str())
+                .unwrap_or_default(),
+        )
+        .unwrap_or_default();
+        let mut request_headers: HashMap<String, Values> =
+            if let ForwardValues::None = &self.send_headers {
+                HashMap::new()
+            } else {
+                serde_json::from_str::<HashMap<String, Vec<String>>>(&headers)
+                    .unwrap_or_default()
+                    .into_iter()
+                    .map(|(header_name, value)| (header_name.to_lowercase(), Values { value }))
+                    .collect()
+            };
+
+        match &self.send_headers {
+            ForwardValues::Only(headers_to_keep) => {
+                request_headers.retain(|header_name, _v| headers_to_keep.contains(header_name));
+            }
+            ForwardValues::Except(skip_headers) => {
+                request_headers.retain(|header_name, _v| !skip_headers.contains(header_name));
+            }
+            ForwardValues::None | ForwardValues::All => {}
+        }
 
-        Ok(Exporter::new(collector, graph))
+        Http {
+            method,
+            host: uri.host().map(|h| h.to_string()).unwrap_or_default(),
```

Comment:
We don't send host or path from Apollo Server and Studio doesn't actually use it. (We found that the ability to figure out what URL you're running at was actually pretty unreliable based on how your server is set up, proxies, etc.)

Reply:
Actually let me have a firmer recommendation here. I think you should *not* send the URL pieces like host and path unless we have a concrete end-to-end feature use case for them. That's because they are an opportunity to accidentally send PII, if for some reason the customer's setup is something like "you POST to `/graphql?userId=foo`". We should err on the side of not including string fields that come from the end user rather than the schema.

Reply:
Fixed in [e17c5d0](https://github.com/apollographql/router/pull/1514/commits/e17c5d00455ba0fa7cb40a9dfaa752b49ad1ee42)